_pytest/test_sanity.py

```diff
@@ -36,7 +33,23 @@ def import_submodules(package_name, skip_list):
 def test_no_global_imports_of_banned_package(banned_package):
     """This test ensures that neither of the banned packages are imported module wise in any of our code files.
 
-    If one of the dependencies is needed, they should be imported within a function."""
+     If one of the dependencies is needed, they should be imported within a function."""
+
+    q = Queue()
+    p = Process(target=get_tracked_imports, args=(q,))
```

Comment:
The reload function in `import_submodules` was necessary as for each parameter of `test_no_global_imports_of_banned_package` the same modules were imported in the loop. On the first pass they were loaded from the OS filesystem but then as they were in cache, the import call just looked in memory and the test `is_pkg` failed where it shouldn't as these imported objects didn't have a `__path__` attribute. The `reload` call forced the interpreter to look again for these module from the filesystem. But it is a brittle function with many unwanted side effects. Using multiprocessing allow to isolate the interpreter imports and load them from the filesystem each time without having to call `reload`